src/main/java/org/opensearch/ad/NodeStateManager.java

```diff
@@ -235,24 +259,38 @@ public void maintenance() {
         maintenance(states, stateTtl);
     }
 
-    public boolean isMuted(String nodeId) {
-        return backpressureMuter.containsKey(nodeId) && backpressureMuter.get(nodeId).isMuted();
+    public boolean isMuted(String nodeId, String detectorId) {
+        Map<String, BackPressureRouting> routingMap = backpressureMuter.get(detectorId);
+        if (routingMap == null || routingMap.isEmpty()) {
+            return false;
+        }
+        BackPressureRouting routing = routingMap.get(nodeId);
+        return routing != null && routing.isMuted();
     }
 
     /**
      * When we have a unsuccessful call with a node, increment the backpressure counter.
      * @param nodeId an ES node's ID
+     * @param detectorId Detector ID
      */
-    public void addPressure(String nodeId) {
-        backpressureMuter.computeIfAbsent(nodeId, k -> new BackPressureRouting(k, clock, settings)).addPressure();
+    public void addPressure(String nodeId, String detectorId) {
+        Map<String, BackPressureRouting> routingMap = backpressureMuter
+            .computeIfAbsent(detectorId, k -> new HashMap<String, BackPressureRouting>());
+        routingMap.computeIfAbsent(nodeId, k -> new BackPressureRouting(k, clock, maxRetryForUnresponsiveNode, mutePeriod)).addPressure();
     }
 
     /**
      * When we have a successful call with a node, clear the backpressure counter.
      * @param nodeId an ES node's ID
+     * @param detectorId Detector ID
      */
-    public void resetBackpressureCounter(String nodeId) {
-        backpressureMuter.remove(nodeId);
+    public void resetBackpressureCounter(String nodeId, String detectorId) {
+        Map<String, BackPressureRouting> routingMap = backpressureMuter.get(detectorId);
+        if (routingMap == null || routingMap.isEmpty()) {
+            backpressureMuter.remove(detectorId);
+            return;
+        }
```

Comment:
why only remove if `routingMap` is empty?

Reply:
if routingMap is not empty, there can be other nodes' mapping from node id to BackPressureRouting.  For example, current mapping is {node1->BackPressureRouting1, node2->BackPressureRouting2}.  The method arguments are (node1, detectorId).  We don't want to remove everything when the client just wants to remove node1. In this example, 

backpressureMuter is a map of map: {detectorId -> {node1->BackPressureRouting1, node2->BackPressureRouting2}}, while routingMap is a map  {node1->BackPressureRouting1, node2->BackPressureRouting2}.

Reply:
Got it. What about the case of a `routingMap` containing a single mapping that will be deleted based on the passed-in `nodeId`. Shall we remove it from `backpressureMuter` as well?

Example: call the method with `node1`, and `backpressureMuter` is {detectorId -> {node1->BackPressureRouting1}}.

Can we move the `routingMap.remove(nodeId)` before doing the empty/null check, so if there is one entry it will clear everything at once?

Reply:
Have to do an empty/null check before "routingMap.remove(nodeId);" as if empty/null check is true, we don't need to do "routingMap.remove(nodeId);".

In your example, after executing "routingMap.remove(nodeId);", routingMap will become empty.  Since we call resetBackpressureCounter in every detector interval, the whole routingMap will be removed in the next interval.

I can do one more empty/null check  after "routingMap.remove(nodeId)", but this seems wordy.  Since we will do it in the next interval, it seems fine.  Let me know what you think.

Reply:
Understood. Seems unnecessary to rewrite it in a way to clean up everything in one go if it is getting cleaned on the next interval.